src/providers/contacts/provider.tsx

```diff
@@ -0,0 +1,53 @@
+import { searchContacts } from '@/api/requests/contact'
+import { Contact } from '@/api/types/contact'
+import { usePikeEnabled } from '@/hooks/useFeatureFlags'
+import { FC, PropsWithChildren, createContext, useCallback, useEffect, useMemo, useState } from 'react'
+
+type ContactsContextValue = {
+  contacts: Contact[] | null
+  refresh: () => void
+  loading: boolean
+  error: boolean
+}
+
+export const ContactsContext = createContext<ContactsContextValue>(null as never)
+
+export const ContactsProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState(false)
+  const [contacts, setContacts] = useState<Contact[] | null>(null)
+  const enabled = usePikeEnabled()
+
+  const load = useCallback(async () => {
+    if (!enabled) {
+      return
+    }
+    setLoading(true)
+    try {
+      const contacts = await searchContacts()
+      setContacts(contacts)
+    } catch {
+      setContacts(null)
+      setError(true)
+    } finally {
+      setLoading(false)
+    }
+  }, [enabled])
+
+  const refresh = useCallback(() => {
```

Comment:
If you change app state you will reduce amount of requests to the backend. This is a usual approach with react apps

Reply:
What do you actually suggesting. I'm not getting your point.